### PR TITLE
Set Setpgid on child git processes (#19865)

### LINKED ISSUE #19077: Defunct git processes when running Gitea as PID 1 on docker
### Gitea Version

1.15.7

### Git Version

2.34.1

### Operating System

OpenSuse Leap 15.3 (Docker) 

### How are you running Gitea?

I used an rpm built using the repo at https://build.opensuse.org/package/show/devel:tools:scm/gitea and installed with zypper.
I'm running in a docker container based on OpenSuse Leap 15.3, base os also OpenSuse Leap 15.3

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Description

I have zombie processes whose start time coincides with log entries such as below:
Upon updating my github login credentials, the zombie processes seem to no longer occur.
#13987 is documented as being fixed by #14006 in the 1.14.0 milestone, so I'm reporting this as a possible regression. I'm sorry I haven't been able to upgrade to the latest version to check if the issue still exists. If you need any additional information, please let me know.

```
2022/03/14 15:22:27 ...irror/mirror_pull.go:176:runSync() [E] Failed to update mirror repository &{21 2 Towa <nil> purchasevisualization PurchaseVisualization 購入品管理  2 https://:xxxxxx@github.com/Towa-Japan/PurchaseVisualization.git main 4 0 1 0 0 0 0 0 0 0 0 0 0 0 0 false false false true <nil> [] 0 map[] map[] [] <nil> false 0 <nil> false 0 <nil> 103641 <nil> <nil> false false [] default  1610340108 1647238881}:
        Stdout: Fetching origin

        Stderr: remote: Repository not found.
        fatal: Authentication failed for 'https://github.com/Towa-Japan/PurchaseVisualization.git/'
        error: Could not fetch origin

        Err: exit status 1
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by zeripath**
Backport #19865

When Gitea is running as PID 1 git will occassionally orphan child processes leading
to (defunct) processes. This PR simply sets Setpgid to true on these child processes
meaning that these defunct processes will also be correctly reaped.

Fix #19077

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: modules/ssh/ssh.go [hunk 1]
```diff
@@ -100,6 +100,8 @@ func sessionHandler(session ssh.Session) {
 	}
 	defer stdin.Close()
 
+	process.SetSysProcAttribute(cmd)
```
Comment (techknowlogick) [inline modules/ssh/ssh.go:104]: CI fail is relevant, `code.gitea.io/gitea/modules/process` will need to be imported in this file

Comment (zeripath) [reply modules/ssh/ssh.go:104]: Ugh not sure how vscode and thence I missed that. I definitely fixed at least one other place.

Comment (zeripath) [reply modules/ssh/ssh.go:104]: Ah I did fix it - I just didn't add it to the commit.

File: modules/process/manager_unix.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+//go:build !windows
+
```
Comment (techknowlogick) [inline modules/process/manager_unix.go:7]:
```suggestion
// +build !windows

```

File: modules/process/manager_windows.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+//go:build windows
+
```
Comment (techknowlogick) [inline modules/process/manager_windows.go:7]:
```suggestion
// +build windows

```

**Review by techknowlogick — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**